### PR TITLE
GROK-15189: EDA: PLS update

### PR DESCRIPTION
**Pull request by vmakarichev**
Multivariate Analysis using PLS is updated:
1. Added computation of PLS components;
2. In the main app:
- added computing explained variances
- provided interactivity
- added circles (score scatter plot)
3. Updated demo app

### CODE REVIEW

File: packages/EDA/src/pls/pls-tools.ts [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+    regressionCoefficients: result[WASM_OUTPUT_IDX.REGR_COEFFS],
+    tScores: result[WASM_OUTPUT_IDX.T_SCORES],
+    uScores: result[WASM_OUTPUT_IDX.U_SCORES],
+    xLoadings: result[WASM_OUTPUT_IDX.X_LOADINGS],
+    yLoadings: result[WASM_OUTPUT_IDX.Y_LOADINGS],
+  };
+}
+
+/** Perform multivariate analysis using the PLS regression */
+async function performMVA(input: PlsInput, analysisType: PLS_ANALYSIS): Promise<void> {
```
Comment (skalkin) [inline packages/EDA/src/pls/pls-tools.ts:62]: Not sure if I like the `analysisType: PLS_ANALYSIS` idea from the architecture / function decomposition point of view - in my experience, it's easier to have three separate functions. But it's up to you :)

File: packages/EDA/src/pls/pls-tools.ts [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+  // 5.1) computation, source: the paper https://doi.org/10.1002/cem.2589
+  //      here, we use notations from this paper
+  const q = result.yLoadings.getRawData();
+  const p = result.xLoadings.map((col) => col.getRawData());
+  const n = input.table.rowCount;
+  const m = featuresNames.length;
+  const A = input.components;
+  const yExplVars = new Float32Array(A);
+  const compNames = [] as string[];
+  const xExplVars = [] as Float32Array[];
```
Comment (skalkin) [inline packages/EDA/src/pls/pls-tools.ts:207]: I just wonder if this notation would be slightly better? Probably no difference, but here you specify the type explicitly vs inferring it. I don't have a strong opinion on that, but decided to mention the other way :)

const xExplVars: Float32Array[] = [];

**Review by skalkin — APPROVED** [review]
Looks great to me! 
Just a few minor comments inside

File: packages/EDA/src/pls/pls-tools.ts [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+    labels: TITLE.FEATURE,
+    help: LINK.LOADINGS,
+  }));
+
+  // 4. Scores Scatter Plot
+
+  // 4.1) data
+  const scoreNames = [] as string[];
+
+  plsCols.forEach((col, idx) => {
```
[the remark this answers is not in the document]
Comment (vmakarichev) [reply packages/EDA/src/pls/pls-tools.ts:134]: Sure! Done!

File: packages/EDA/src/pls/pls-tools.ts (hunk 2 above)
Comment (vmakarichev) [reply packages/EDA/src/pls/pls-tools.ts:207]: Modified! Your suggestion is more preferable.

File: packages/EDA/src/pls/pls-tools.ts (hunk 1 above)
Comment (vmakarichev) [reply packages/EDA/src/pls/pls-tools.ts:62]: Let's use a single function

File: packages/EDA/src/pls/pls-tools.ts [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+  scoreNames.forEach((xName) => {
+    const x = '${' + xName + '}';
+    lines.push({type: 'line', formula: `${x} = 0`, width: LINE_WIDTH, visible: true, title: ' ', color: COLOR.AXIS});
+
+    scoreNames.forEach((yName) => {
+      const y = '${' + yName + '}';
+
+      RADIUS.forEach((r) => {
+        const formula1 = y + ` = sqrt(${r*r} - ${x} * ${x})`;
+        lines.push({
```
[the remark this answers is not in the document]
Comment (vmakarichev) [reply packages/EDA/src/pls/pls-tools.ts:167]: Done

**Pull request merged**